src/server/listener.cc

```diff
@@ -370,23 +365,22 @@ void Listener::dispatchPeer(const std::shared_ptr<Peer> &peer) {
 
 #ifdef PISTACHE_USE_SSL
 
-static SSL_CTX *ssl_create_context(const std::string &cert,
-                                   const std::string &key,
-                                   bool use_compression) {
-  const SSL_METHOD *method;
-  SSL_CTX *ctx;
+namespace {
 
-  method = SSLv23_server_method();
+ssl::SSLCtxPtr ssl_create_context(const std::string &cert,
+                                         const std::string &key,
+                                         bool use_compression) {
+  const SSL_METHOD* method = SSLv23_server_method();
 
-  ctx = SSL_CTX_new(method);
-  if (ctx == NULL) {
+  ssl::SSLCtxPtr ctx{SSL_CTX_new(method)};
+  if (ctx == nullptr) {
     ERR_print_errors_fp(stderr);
     throw std::runtime_error("Cannot setup SSL context");
   }
 
   if (!use_compression) {
     /* Disable compression to prevent BREACH and CRIME vulnerabilities. */
-    if (!SSL_CTX_set_options(ctx, SSL_OP_NO_COMPRESSION)) {
+    if (!SSL_CTX_set_options((SSL_CTX *)ctx.get(), SSL_OP_NO_COMPRESSION)) {
```

Comment:
Please do not use legacy C style casts; they will show up as warnings in cppcheck.  Since this one gets casted alot, consider writing a "static inline SSL_CTX* get_context(void *foo)..." that hides the cast in one place (and use a C++ style cast instead).

Reply:
@dennisjenkins75 Thank you for this remark, I totally agree with you.